auth/src/swig/auth.i

```diff
@@ -1179,6 +1182,124 @@ static CppInstanceManager<Auth> g_auth_instances;
+  ///
+  /// Data from the Identity Provider used to sign-in is returned in the
+  /// @ref AdditionalUserInfo inside @ref SignInResult.
+  [System.Obsolete("Please use `Task<AuthResult> LinkWithProviderAsync(Credential)` instead", false)]
+  public System.Threading.Tasks.Task<SignInResult> LinkAndRetrieveDataWithCredentialAsync_DEPRECATED(
+      Credential credential) {
+    ThrowIfNull();
+    var taskCompletionSource =
+        new System.Threading.Tasks.TaskCompletionSource<SignInResult>();
+    LinkAndRetrieveDataWithCredentialInternalAsync_DEPRECATED(credential).ContinueWith(task => {
+        CompleteSignInResultTask(task, taskCompletionSource);
+      });
+    return taskCompletionSource.Task;
+  }
+
+  /// @deprecated This method is deprecated in favor of methods that return
+  /// `Task<AuthResult>`. Please use
+  /// @ref LinkWithCredentialAsync(Credential) instead.
+  ///
+  /// Associates a user account from a third-party identity provider.
+  [System.Obsolete("Please use `Task<AuthResult> LinkWithCredentialAsync(Credential)` instead", false)]
+  public System.Threading.Tasks.Task<FirebaseUser> LinkWithCredentialAsync_DEPRECATED(
+      Credential credential) {
+    ThrowIfNull();
+    var taskCompletionSource =
+        new System.Threading.Tasks.TaskCompletionSource<FirebaseUser>();
+    LinkWithCredentialInternalAsync_DEPRECATED(credential).ContinueWith(
+        task => {
+          if(authProxy != null) {
+              authProxy.CompleteFirebaseUserTask(task, taskCompletionSource);
+          }
+        });
+    return taskCompletionSource.Task;
+  }
+
+  /// @deprecated This method is deprecated in favor of methods that return
+  /// `Task<AuthResult>`. Please use
+  /// @ref ReauthenticateAndRetrieveDataAsync(Credential) instead.
+  ///
+  /// Reauthenticate using a credential.
+  ///
+  /// Data from the Identity Provider used to sign-in is returned in the
+  /// AdditionalUserInfo inside the returned SignInResult.
+  ///
+  /// Returns an error if the existing credential is not for this user
+  /// or if sign-in with that credential failed.
+  ///
+  /// @note: The current user may be signed out if this operation fails on
+  /// Android and desktop platforms.
+  [System.Obsolete("Please use `Task<AuthResult> LinkWithCredentialAsync(Credential)` instead", false)]
+  public System.Threading.Tasks.Task<SignInResult> ReauthenticateAndRetrieveDataAsync_DEPRECATED(
+      Credential credential) {
+    ThrowIfNull();
+    var taskCompletionSource =
+        new System.Threading.Tasks.TaskCompletionSource<SignInResult>();
+    ReauthenticateAndRetrieveDataInternalAsync_DEPRECATED(credential).ContinueWith(task => {
+        CompleteSignInResultTask(task, taskCompletionSource);
+      });
+    return taskCompletionSource.Task;
+  }
+
+  /// @deprecated This method is deprecated in favor of methods that return
+  /// `Task<AuthResult>`. Please use
+  /// @ref UnlinkAsync(string) instead.
```

Comment:
nit: there's enough line space on 1260 to fit the rest of this text, I think.

Reply:
Just easier to copy-paste. Done.